File: app/src/main/java/sgtmelon/handynotes/model/state/SortState.java
package sgtmelon.handynotes.model.state;

import java.util.List;

import sgtmelon.handynotes.model.item.ItemSort;
import sgtmelon.handynotes.service.Help;

public class SortState {

    private int end;

    public int getEnd() {
        return end;
    }

    public void updateEnd(List<ItemSort> listSort) {
        for (int k = 0; k < listSort.size(); k++) {
            int key = listSort.get(k).getKey();
            if (key == Help.Pref.sortCr || key == Help.Pref.sortCh) {
                end = k;
                break;
            }
        }
    }

}

File: app/src/main/java/sgtmelon/scriptum/app/model/state/SortState.kt
package sgtmelon.scriptum.app.model.state

import sgtmelon.scriptum.app.model.item.SortItem
import sgtmelon.scriptum.office.annot.def.SortDef

/**
 * Состояние для сортировки, определяющее позицию элемента [SortItem], до которого будет происходить сортировка
 */
class SortState {

    var end: Int = 0
        private set

    fun updateEnd(listSort: List<SortItem>) {
        for (i in listSort.indices) {
            val key = listSort[i].key

            if (key == SortDef.create || key == SortDef.change) {
                end = i
                break
            }
        }
    }

}